templates/example_rule.py

```diff
@@ -0,0 +1,33 @@
+## Required Functions
+def rule(event): # the logic for setting off an alert, return True = Alert, False = Do not Alert
+    if event.get('Something'):
+        return True
+    return False
+
+## Optional Functions
+def title(event): # User can set custom alert titles in string format
+    return 'This is my title'
+
+def dedup(event): # by default if not defined the deduplication period string is based off the alert title 
+    return 'DedupString'
+def alert_context(event): # additional information that the user may find useful that is passed in with the alert as a dictionary
+    return dict(event)
+
+## Override Functions
```

Comment:
There is no real distinction to Panther on what you label `Optional Functions` and `Override Functions`. In both cases they are using python to override YAML settings in the rule for a particular alert. We can distinguish here if you think that's useful, but functionally they're doing the same thing.

Reply:
Adding a Note at the the top to clarify this